sql/src/test/java/org/apache/druid/sql/calcite/WindowQueryTestBase.java

```diff
@@ -0,0 +1,379 @@
+    }
+
+    @Override
+    public SpecificSegmentsQuerySegmentWalker createQuerySegmentWalker(
+        QueryRunnerFactoryConglomerate conglomerate,
+        JoinableFactoryWrapper joinableFactory,
+        Injector injector
+    )
+    {
+      final SpecificSegmentsQuerySegmentWalker retVal = super.createQuerySegmentWalker(
+          conglomerate,
+          joinableFactory,
+          injector);
+
+      final File tmpFolder = tempDirProducer.newTempFolder();
+      TestDataBuilder.attachIndexesForDrillTestDatasources(retVal, tmpFolder);
+      return retVal;
+    }
+  }
+
+  private class TextualResultsVerifier implements ResultsVerifier
+  {
+    protected final List<String[]> expectedResultsText;
+    @Nullable
+    protected final RowSignature expectedResultRowSignature;
+    private RowSignature currentRowSignature;
+
+    public TextualResultsVerifier(List<String[]> expectedResultsString, RowSignature expectedSignature)
+    {
+      this.expectedResultsText = expectedResultsString;
+      this.expectedResultRowSignature = expectedSignature;
+    }
+
+    @Override
+    public void verifyRowSignature(RowSignature rowSignature)
+    {
+      if (expectedResultRowSignature != null) {
+        Assert.assertEquals(expectedResultRowSignature, rowSignature);
+      }
+      currentRowSignature = rowSignature;
+    }
+
+    @Override
+    public void verify(String sql, QueryTestRunner.QueryResults queryResults)
+    {
+      List<Object[]> results = queryResults.results;
+      List<Object[]> expectedResults = parseResults(currentRowSignature, expectedResultsText);
+      try {
+        Assert.assertEquals(StringUtils.format("result count: %s", sql), expectedResultsText.size(), results.size());
+        if (!isOrdered(queryResults)) {
+          // in case the resultset is not ordered; order via the same comparator before comparison
+          results.sort(new ArrayRowCmp());
+          expectedResults.sort(new ArrayRowCmp());
+        }
+        assertResultsValid(ResultMatchMode.EQUALS_RELATIVE_1000_ULPS, expectedResults, queryResults);
+      }
+      catch (AssertionError e) {
+        log.info("query: %s", sql);
+        log.info(resultsToString("Expected", expectedResults));
+        log.info(resultsToString("Actual", results));
+        throw new AssertionError(StringUtils.format("%s while processing: %s", e.getMessage(), sql), e);
+      }
+    }
+
+    private boolean isOrdered(QueryTestRunner.QueryResults queryResults)
+    {
+      SqlNode sqlNode = queryResults.capture.getSqlNode();
+      return SqlToRelConverter.isOrdered(sqlNode);
+    }
+  }
+
+  private static class ArrayRowCmp implements Comparator<Object[]>
+  {
+    @Override
+    public int compare(Object[] arg0, Object[] arg1)
+    {
+      String s0 = Arrays.toString(arg0);
+      String s1 = Arrays.toString(arg1);
+      return s0.compareTo(s1);
+    }
+  }
+
+  private static List<Object[]> parseResults(RowSignature rs, List<String[]> results)
```

Comment:
Have reverted the newly added test class, and have added the new tests in DrillWindowQueryTest itself. Have kept this as a [separate commit](https://github.com/apache/druid/pull/16729/commits/002bf1f54c3ba5d3c9acbb14735f0e8b88422416), in case we want to revert this post further discussion,.
Hope that works.